kythe/go/platform/tools/viewindex/viewindex.go

```diff
@@ -61,6 +66,61 @@ func main() {
 	}
 
 	path := flag.Arg(0)
+	if !strings.HasSuffix(path, ".kzip") {
+		// Assume deprecated .kindex, by default
+		viewKIndex(path)
+		return
+	}
+
+	f, err := os.Open(path)
+	if err != nil {
+		log.Fatalf("Error opening %q: %v", path, err)
+	}
+	defer f.Close()
+
+	info, err := f.Stat()
+	if err != nil {
+		log.Fatalf("Error getting size of %q: %v", path, err)
+	}
+
+	rd, err := kzip.NewReader(f, info.Size())
+	if err != nil {
+		log.Fatalf("Error opening kzip reader %q: %v", path, err)
+	}
+
+	if *printFile != "" {
+		data, err := rd.ReadAll(*printFile)
+		if err != nil {
+			log.Fatalf("Error reading file with digest %q: %v", *printFile, err)
+		} else if _, err := os.Stdout.Write(data); err != nil {
+			log.Fatal(err)
+		}
+		return
+	}
+
+	out := os.Stdout
+	en := json.NewEncoder(out)
+	if err := rd.Scan(func(u *kzip.Unit) error {
+		if !*printFiles {
+			return m.Marshal(out, u.Proto)
+		}
+
+		idx := &kindex.Compilation{Proto: u.Proto}
+		for _, f := range u.Proto.RequiredInput {
+			data, err := rd.ReadAll(f.Info.GetDigest())
+			if err != nil {
+				return fmt.Errorf("reading file digest %q: %v", f.Info.GetDigest(), err)
+			}
+			idx.Files = append(idx.Files, &apb.FileData{Content: data, Info: f.Info})
+		}
+
+		return en.Encode(idx)
+	}); err != nil {
+		log.Fatalf("Error scanning kzip %q: %v", path, err)
+	}
+}
+
+func viewKIndex(path string) {
```

Comment:
the idea of not having viewZip / viewKIndex is that the latter is going to go away sometime?  If so that sounds ok, otherwise maybe have separate functions

Reply:
Yeah, .kindex is going to be removed.